### PR TITLE
feat: 스프링 시큐리티, jwt 토큰 설정

### PR DESCRIPTION
**Pull request by bokyung95**
### **변경 사항**
**AS-IS**
스프링 시큐리티
jwt 토큰 설정
회원가입, 로그인

**TO-BE**
validation, 예외처리
회원가입, 로그인 테스트 코드 작성

### **테스트**
- [ ]  테스트 코드
- [V]  API 테스트

### CODE REVIEW

File: src/main/java/com/gang/oneclickshoppingmall/jwt/JwtTokenProvider.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	private String secretKey;
+
+	private final long TOKEN_VALID_TIME = 30 * 60 * 1000L; //30분
+
+	private final UserDetailsService userDetailsService;
+
+	/**
+	 * 객체 초기화, secretKey -> Base64로 인코딩
+	 */
+	protected void init() {
```
Comment (naekang) [inline src/main/java/com/gang/oneclickshoppingmall/jwt/JwtTokenProvider.java:33]: 객체 초기화 동작을 하기 위한 `@PostConstructor` 어노테이션이 생략되어 있는 것 같습니다 :)

File: src/main/java/com/gang/oneclickshoppingmall/member/service/MemberService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+
+	private final MemberRepository memberRepository;
+	private final PasswordEncoder bCryptPasswordEncoder;
+	private final JwtTokenProvider jwtTokenProvider;
+
+	public Long registerMember(SignUpRequestDto requestDto) {
```
Comment (naekang) [inline src/main/java/com/gang/oneclickshoppingmall/member/service/MemberService.java:25]: 회원 가입 시 이미 가입된 이메일이 있는지 확인하는 예외 처리도 있으면 좋을거같습니다!

File: src/main/java/com/gang/oneclickshoppingmall/member/type/MemberStatus.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.gang.oneclickshoppingmall.member.type;
+
+public enum MemberStatus {
+	ACTIVE, DORMANT, INACTIVE
```
Comment (naekang) [inline src/main/java/com/gang/oneclickshoppingmall/member/type/MemberStatus.java:4]: 새로운 영어단어 배워갑니다... 👍

**Review by naekang — COMMENTED** [review]
👍 GOD 👍

**Comment by bokyung95** [comment]
|File|Coverage [22.83%]|:x:|
|:-|:-:|:-:|
|[MemberStatus.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Ftype%2FMemberStatus.java)|100%|:green_apple:|
|[MemberRole.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Ftype%2FMemberRole.java)|100%|:green_apple:|
|[SecurityConfig.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fconfig%2FSecurityConfig.java)|100%|:green_apple:|
|[OneclickShoppingmallApplication.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2FOneclickShoppingmallApplication.java)|37.5%|:x:|
|[MemberController.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fcontroller%2FMemberController.java)|33.33%|:x:|
|[BaseEntity.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fbase%2Fentity%2FBaseEntity.java)|33.33%|:x:|
|[SecurityUserDetailService.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fjwt%2FSecurityUserDetailService.java)|25%|:x:|
|[MemberService.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fservice%2FMemberService.java)|19.67%|:x:|
|[JwtAuthenticationFilter.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fjwt%2FJwtAuthenticationFilter.java)|19.35%|:x:|
|[JwtTokenProvider.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fjwt%2FJwtTokenProvider.java)|16.83%|:x:|
|[Member.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fentity%2FMember.java)|1.63%|:x:|
|[ErrorCode.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fexception%2Ftype%2FErrorCode.java)|0%|:x:|
|[CustomException.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fexception%2Ftype%2FCustomException.java)|0%|:x:|
|[SignUpDto.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fdto%2FSignUpDto.java)|0%|:x:|
|[LoginDto.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fdto%2FLoginDto.java)|0%|:x:|
|[MemberDetails.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fmember%2Fentity%2FMemberDetails.java)|0%|:x:|
|[CustomException.java](https://github.com/bokyung95/oneclick/blob/f329fcbb51e1bac58f44aa972595f01c2bb3dd92/src%2Fmain%2Fjava%2Fcom%2Fgang%2Foneclickshoppingmall%2Fexception%2FCustomException.java)|0%|:x:|

|Total Project Coverage|22.83%|:x:|
|:-|:-:|:-:|

**Pull request merged**